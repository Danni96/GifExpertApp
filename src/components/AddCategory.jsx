import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("Solo leveling");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <>
      <form onSubmit={onSubmit}>
        <input
          type="text"
          value={inputValue}
          placeholder="Buscar gifs"
          onChange={onInputChange}
        />
      </form>
    </>
  );

  // validar la props de setcategories
};

// AddCategory.propTypes = {
//   onNewCategory: PropTypes.string.isRequired,
// };
