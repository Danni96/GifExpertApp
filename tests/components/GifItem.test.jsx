import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("Pruebas en <GifItem />", () => {
  const title = "Pruebas en";
  const url = "http://hola.test/images.png";

  test("debe hacer match con el snap", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  
  test("debe mostrar la imagen con el url y el ALT", () => {
    render(<GifItem title={title} url={url} />);
    // expect(screen.getByRole('img').src).toBe(url);

    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });



});
