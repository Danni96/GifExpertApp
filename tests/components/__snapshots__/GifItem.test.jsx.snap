// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snap 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pruebas en"
      src="http://hola.test/images.png"
    />
    <p>
      Pruebas en
    </p>
  </div>
</div>
`;
